gpytorch/variational/ciq_variational_strategy.py

```diff
@@ -0,0 +1,238 @@
+            max_tridiag_iter=settings.max_lanczos_quadrature_iterations.value(),
+            preconditioner=lambda x: x / diag,
+        )
+        expec_vec = solves[..., 0]
+        s_times_interp_term = solves[..., 1:]
+
+        # Compute the interpolated mean
+        # k^T K^{-1/2} m
+        interp_mean = (s_times_interp_term.transpose(-1, -2) @ natural_vec.unsqueeze(-1)).squeeze(-1)
+
+        # Compute the interpolated variance
+        # k^T K^{-1/2} S K^{-1/2} k = k^T K^{-1/2} (expec_mat - expec_vec expec_vec^T) K^{-1/2} k
+        interp_var = (s_times_interp_term * interp_term).sum(dim=-2)
+
+        # Let's not bother actually computing the KL-div in the foward pass
+        # 1/2 ( -log | S | + tr(S) + m^T m - len(m) )
+        # = 1/2 ( -log | expec_mat - expec_vec expec_vec^T | + tr(expec_mat) - len(m) )
+        kl_div = torch.zeros_like(interp_mean[..., 0])
+
+        # We're done!
+        ctx.save_for_backward(interp_term, s_times_interp_term, interp_mean, natural_vec, expec_vec, prec)
+        return interp_mean, interp_var, kl_div
+
+    @staticmethod
+    def backward(ctx, interp_mean_grad, interp_var_grad, kl_div_grad):
+        # Get the saved terms
+        interp_term, s_times_interp_term, interp_mean, natural_vec, expec_vec, prec = ctx.saved_tensors
+
+        # Expand data-depenedent gradients
+        interp_mean_grad = interp_mean_grad.unsqueeze(-2)
+        interp_var_grad = interp_var_grad.unsqueeze(-2)
+
+        # Compute gradient of interp term (K^{-1/2} k)
+        # interp_mean component: m
+        # interp_var component: S K^{-1/2} k
+        # kl component: 0
+        interp_term_grad = (interp_var_grad * s_times_interp_term).mul(2.0) + (
+            interp_mean_grad * expec_vec.unsqueeze(-1)
+        )
+
+        # Compute gradient of expected vector (m)
+        # interp_mean component: K^{-1/2} k
+        # interp_var component: (k^T K^{-1/2} m) K^{-1/2} k
+        # kl component: S^{-1} m
+        expec_vec_grad = sum(
+            [
+                (interp_var_grad * interp_mean.unsqueeze(-2) * interp_term).sum(dim=-1).mul(-2),
+                (interp_mean_grad * interp_term).sum(dim=-1),
+                (kl_div_grad.unsqueeze(-1) * natural_vec),
+            ]
+        )
+
+        # Compute gradient of expected matrix (mm^T + S)
+        # interp_mean component: 0
+        # interp_var component: K^{-1/2} k k^T K^{-1/2}
+        # kl component: 1/2 ( I - S^{-1} )
+        eye = torch.eye(expec_vec.size(-1), device=expec_vec.device, dtype=expec_vec.dtype)
+        expec_mat_grad = torch.add(
+            (interp_var_grad * interp_term) @ interp_term.transpose(-1, -2),
+            (kl_div_grad.unsqueeze(-1).unsqueeze(-1) * (eye - prec).mul(0.5)),
+        )
+
+        # We're done!
+        return interp_term_grad, expec_vec_grad, expec_mat_grad, None  # Extra "None" for the kwarg
+
+
+class VD(Module):
```

Comment:
`_CIQNaturalVariationalDistribution`?

Reply:
I'm changing up CIQVariationalStrategy to accept any type of variational distribution - so it should be called with the standard NaturalVariationalDistribution instead.